dask/array/core.py

```diff
@@ -1184,6 +1195,65 @@ def from_func(func, shape, dtype=None, name=None, args=(), kwargs={}):
     return Array(dsk, name, chunks, dtype)
 
 
+def common_blockdim(blockdims):
+    """ Find the common block dimensions from the list of block dimensions
+
+    Currently only implements the simplest possible heuristic: the common
+    block-dimension is the only one that does not span fully span a dimension.
+    This is a conservative choice that allows us to avoid potentially very
+    expensive rechunking.
+
+    Assumes that each element of the input block dimensions has all the same
+    sum (i.e., that they correspond to dimensions of the same size).
+
+    Examples
+    --------
+
+    >>> common_blockdim([(3,), (2, 1)])
+    {(2, 1),}
+    >>> common_blockdim([(2, 2), (3, 1)])  # doctest: +SKIP
+    ValueError: Chunks do not align
```

Comment:
unfortunately, can't do that in this case because the error message is not deterministic (it includes the repr of a set object)